### DISCUSSION TITLE
Dedicated metadata for difficulty creator

### DISCUSSION THREAD

**Discussion by Craftplacer** [ppy/osu#13081] [Ideas]
**Describe the new feature:** 
Currently osu! beatmaps have only one creator, while multiple people have worked on it. My proposal is to split it up into two metadata: beatmap (set) creator, and beatmap/difficulty creator.

**Proposal designs of the feature:** 
Should be obvious. *happy to elaborate if still unclear*

**Backwards compatibility**
Every beatmap / difficulty having this pseudo-regex "*'s *" will get the automatic conversion.

**Example Code & Additional Condition:**
```cs
var values = beatmapCreator.Split("'s ", StringSplitOptions.RemoveEmptyEntries);
var doConversion = values.Length == 2; //makes sure it isn't intentional by the user.
```

**Example Test Case:**
Artist: ゆいこんぬ ＆ あやぽんず＊
Title: すーぱーぬこわーるど
Creator: AllStar12
Version: Guy's Extra

_will get split up into_

Beatmap (Set) Creator: AllStar12
Difficulty Creator: Guy
Version: Extra

I apologize in advance for the poor phrasing.

**Comment by camellirite** [comment]
it'd have to be a bit more complex than looking for "'s". some stuff includes:
- collaboration difficulties like ["Collab Hard"](https://osu.ppy.sh/beatmapsets/1028168#osu/2182836) not getting recognized
- combinations of names such as ["Linadeft's Extra"](https://osu.ppy.sh/beatmapsets/617203#osu/1324163)
- special difficulties having "'s" in the name like [Fate's Mark](https://osu.ppy.sh/beatmapsets/853796#osu/1784323) getting picked up falsely

usernames from everyone contributing is required to be in the tags by the ranking criteria so I imagine it's possible to catch 99% of cases, but it's a lot more difficult than just regex for "'s"

**Comment by Craftplacer** [comment]
This could be forced upon at some point to make this a standard.
About the conversion thing, I think there needs to be some special cases. Also this could be run inside the server, so we can detect users too without having to abuse the API.

So, what about a single person worked on this? I think it simply should default to the beatmap set creator.
